parlai/agents/torch/torch.py

```diff
@@ -0,0 +1,257 @@
+# Copyright (c) 2017-present, Facebook, Inc.
+# All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree. An additional grant
+# of patent rights can be found in the PATENTS file in the same directory.
+
+from parlai.core.agents import Agent
+from parlai.core.dict import DictionaryAgent
+
+try:
+    import torch
+except Exception as e:
+    raise ModuleNotFoundError('Need to install Pytorch: go to pytorch.org')
+
+from collections import deque, namedtuple
+import random
+
+Batch = namedtuple("Batch", ["text_vecs", "label_vecs", "labels",
+                             "valid_inds"])
+
+
+class TorchAgent(Agent):
+    """Base Agent for all models which use Torch.
+
+    This agent serves as a common framework for all ParlAI models which want
+    to use PyTorch.
+    """
+
+    @staticmethod
+    def dictionary_class():
+        return DictionaryAgent
+
+    @staticmethod
+    def add_cmdline_args(argparser):
+        agent = argparser.add_argument_group('TorchAgent Arguments')
+        agent.add_argument('-hist', '--history-length', default=10, type=int,
```

Comment:
i think often you want to parameterize on number of utterances in history also.. i guess could be another param

Reply:
default 10 is weird